app/frontend/src/features/communities/events/EventPage.tsx

```diff
@@ -0,0 +1,211 @@
+  const isValidEventId = !isNaN(eventId) && eventId > 0;
+  const queryClient = useQueryClient();
+  const {
+    data: event,
+    error: eventError,
+    isLoading,
+  } = useQuery<Event.AsObject, GrpcError>({
+    queryKey: eventKey(eventId),
+    queryFn: () => service.events.getEvent(eventId),
+    enabled: isValidEventId,
+  });
+
+  const {
+    isLoading: isSetEventAttendanceLoading,
+    error: setEventAttendanceError,
+    mutate: setEventAttendance,
+  } = useMutation<Event.AsObject, GrpcError, AttendanceState>(
+    (currentAttendanceState) => {
+      const attendanceStateToSet =
+        currentAttendanceState === AttendanceState.ATTENDANCE_STATE_GOING
+          ? AttendanceState.ATTENDANCE_STATE_NOT_GOING
+          : AttendanceState.ATTENDANCE_STATE_GOING;
+      return service.events.setEventAttendance({
+        attendanceState: attendanceStateToSet,
+        eventId,
+      });
+    },
+    {
+      onSuccess(updatedEvent) {
+        queryClient.setQueryData<Event.AsObject>(
+          eventKey(eventId),
+          updatedEvent
+        );
+        queryClient.invalidateQueries(eventKey(eventId));
+        queryClient.invalidateQueries({
+          predicate: ({ queryKey }) =>
+            queryKey[0] === eventAttendeesBaseKey &&
+            isEventUsersInputType(queryKey[1]) &&
+            queryKey[1].eventId === eventId,
+        });
+      },
+    }
+  );
+
+  useEffect(() => {
+    if (event?.slug && event.slug !== eventSlug) {
+      history.replace(routeToEvent(event.eventId, event.slug));
+    }
+  }, [event, eventSlug, history]);
+
+  return !isValidEventId ? (
+    <NotFoundPage />
+  ) : (
+    <>
+      {(eventError || setEventAttendanceError) && (
+        <Alert severity="error">
+          {eventError?.message || setEventAttendanceError?.message || ""}
+        </Alert>
+      )}
+      {isLoading ? (
+        <CircularProgress />
+      ) : (
+        event && (
+          <>
+            <div className={classes.header}>
+              <HeaderButton
+                className={classes.backButton}
+                onClick={() => history.goBack()}
+              >
+                <BackIcon />
+              </HeaderButton>
+              <div className={classes.eventTitle}>
+                <Typography variant="h1">{event.title}</Typography>
+                <Typography className={classes.eventTypeText} variant="body1">
+                  {event.onlineInformation
+                    ? VIRTUAL_EVENT
+                    : event.offlineInformation?.address}
+                </Typography>
+              </div>
+              <Button
+                className={classes.attendanceButton}
+                loading={isSetEventAttendanceLoading}
+                onClick={() => setEventAttendance(event.attendanceState)}
+              >
+                {event.attendanceState ===
+                AttendanceState.ATTENDANCE_STATE_GOING
+                  ? NOT_GOING
+                  : GOING}
+              </Button>
```

Comment:
Probably should extend this to a dropdown to include the "Maybe" option later, but I think ok for now for MVP/V1 rollout of this page?